src/components/local-weather.jsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { fetchWeather } from "../utils";
+import Search from "./search";
+import Today from "./today-weather";
+import FiveDay from "./five-day-weather";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+
+class LocalWeather extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { currentCity: props.city, weather: {}, showError: false, units: "imperial", copied: false };
+    
+    this.fetchWeatherData = this.fetchWeatherData.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleCopy = this.handleCopy.bind(this);
+    
+    // default fetch weather with imperial units
+    if (this.props.city) this.fetchWeatherData(this.props.city.split(", ").join(","), "imperial");
+  }
+
+  componentDidUpdate() {
+    // when the user enters new city info in the search bar, refetch the weather
+    if (this.props.city !== this.state.currentCity) {
+      this.fetchWeatherData(this.props.city, this.state.units);
+    }
+  }
+
+  fetchWeatherData(city, unit) {
+    fetchWeather(city, unit).then(data => {
+      // when a city changes and new weather is fetched, also update the currentCity and show link as not copied
+      this.setState({ weather: data, currentCity: this.props.city, copied: false });
+    }).catch(err => {
+      this.setState({showError: true});
+    });
+  }
+
+  handleClick() {
+    // handles switching imperial to metric and vice versa
+    const newUnits = this.state.units === "imperial" ? "metric" : "imperial";
+    this.setState({ units: newUnits });
+    this.fetchWeatherData(this.props.city.split(", ").join(","), newUnits);
+  }
```

Comment:
Looks like you're fetching the weather every time the user changes the units from Imperial to Metric & vice versa. If you were asked to make switching units more efficient, how would you go about that?

Reply:
Rather than making a new call, I would set up a celsius/Fahrenheit converter function and use that when the user clicked the toggle button.